src/serializers/infer.rs

```diff
@@ -83,22 +81,6 @@ pub(crate) fn infer_to_python_known(
         }};
     }
 
-    let serialize_dict = |dict: &PyDict| {
```

Comment:
before we had a closure that required a dict, again requiring an intermediate dict for dataclasses, now we have a function which takes a generic.